backend/stakepoold/server.go

```diff
@@ -368,45 +728,178 @@ func (ctx *appContext) vote(wg *sync.WaitGroup, blockHash *chainhash.Hash, block
 		return
 	}
 
-	// Ask wallet to transmit raw transaction.
+	// Ask node to transmit raw transaction.
 	startSend := time.Now()
-	_, w.err = ctx.nodeConnection.SendRawTransaction(newTx, false)
+	tx, err := ctx.nodeConnection.SendRawTransaction(newTx, false)
+	if err != nil {
+		log.Infof("vote err %v", err)
+		w.err = err
+	} else {
+		w.txid = tx
+	}
 	w.sendDuration = time.Since(startSend)
 }
 
+func (ctx *appContext) processNewTickets(nt NewTicketsForBlock) {
+	time.Sleep(votingTargetTime) // give processWinningTickets time to vote
```

Comment:
Sleep is not the solution; please use a signaling mechanism.  Sleep is the new goto.